### PR TITLE
Expanded tests, policy tests

### PR DESCRIPTION
**Pull request by Cerfoglg**
## Proposed changes

Expanding tests for the API

## Types of changes

What types of changes does your code introduce to the project: _Put
an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
      functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after
creating the PR. If you're unsure about any of them, don't hesitate to
ask. We're here to help! This is simply a reminder of what we are going
to look for before merging your code._

- [x] I have read the [CONTRIBUTING][contrib] doc
- [x] I have signed the [Contributor License Agreement][cla]
- [ ] I have updated the [RELEASE NOTES][release]
- [x] I have added tests that prove my fix is effective or that my
      feature works
- [x] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in
      downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion
by explaining why you chose the solution you did and what alternatives
you considered, etc...

[cla]: https://raw.githubusercontent.com/orchestracities/opa-authz/master/individual_cla.pdf
    "Martel Open Source Software Individual Contributor License Agreement"
[contrib]: https://github.com/orchestracities/opa-authz/blob/master/CONTRIBUTING.md
    "Contributing to OPA-authz"
[release]: https://github.com/orchestracities/opa-authz/blob/master/RELEASE_NOTES.md
    "OPA-authz Release Notes"

### CODE REVIEW

**Comment by chicco785** [comment]
let's test using `-H "Host: keycloak"` with curl api calls to avoid having to set values in `/etc/hosts`

File: scripts/test_context_broker.sh [hunk 1]
```diff
@@ -8,12 +8,22 @@ jq '.access_token'`
...
-curl -i -H "Authorization: Bearer $token" -H "fiware-service: Tenant1" -H "fiware-servicepath: /" 'http://localhost:8000/v2/entities'
+echo "Can i read entities?"
+echo "==============================================================="
+export response=`curl -s -o /dev/null -w "%{http_code}" -H "Authorization: Bearer $token" -H "fiware-service: Tenant1" -H "fiware-servicepath: /" 'http://localhost:8000/v2/entities'`
+if [ $response == "200" ]
+then
+  echo "PASSED"
+else
+  echo "ERROR: Can't read entities"
+  exit 1
```
Comment (chicco785) [inline scripts/test_context_broker.sh:19]: shouldn't be -1?

Comment (chicco785) [reply scripts/test_context_broker.sh:19]: forgive me ignorance on this aspects :)

File: anubis-management-api/src/tests/test_policies.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            "fiware_service": "test",
+            "fiware_service_path": "/"})
+    body = response.json()
+    assert response.status_code == 200
+    assert body["access_to"] == "resource"
+    assert body["resource_type"] == "entity"
+    assert body["mode"] == ["acl:Read"]
+    assert body["agent"] == ["acl:agent:test"]
+
+    response = client.get(
```
[the remark this answers is not in the document]
Comment (chicco785) [reply anubis-management-api/src/tests/test_policies.py:53]: @Cerfoglg incase you missed this comment

File: scripts/test_context_broker.sh (hunk 1 above)
Comment (Cerfoglg) [reply scripts/test_context_broker.sh:19]: Exiting with 1 indicates a general error. There are [other errors too](https://shapeshed.com/unix-exit-codes/), but I don't think -1 is an error.

File: scripts/test_context_broker.sh [hunk 3]
```diff
@@ -134,113 +59,70 @@ curl -i --request POST 'http://localhost:8000/v2/entities' \
...
-echo "\n\ncan i read entity urn:ngsi-ld:AirQualityObserved:demo attributes?"
-curl -i -H "Authorization: Bearer $token" -H "fiware-service: Tenant1" -H "fiware-servicepath: /" http://localhost:8000/v2/entities/urn:ngsi-ld:AirQualityObserved:demo/attrs
+echo "Can I delete entity urn:ngsi-ld:AirQualityObserved:demo?"
+echo "==============================================================="
+export response=`curl -s -o /dev/null -w "%{http_code}" --request DELETE -H "Authorization: Bearer $token" -H "fiware-service: Tenant1" -H "fiware-servicepath: /" http://localhost:8000/v2/entities/urn:ngsi-ld:AirQualityObserved:demo`
+if [ $response == "204" ]
+then
+  echo "PASSED"
+else
+  echo "ERROR: Can create entity"
```
Comment (chicco785) [inline scripts/test_context_broker.sh:124]: can't delete entity?

File: scripts/test_context_broker.sh [hunk 4]
```diff
@@ -134,113 +59,70 @@ curl -i --request POST 'http://localhost:8000/v2/entities' \
...
+echo ""
+
+echo "Can I read entity urn:ngsi-ld:AirQualityObserved:demo temperature?"
+echo "==============================================================="
+export response=`curl -s -o /dev/null -w "%{http_code}" -H "Authorization: Bearer $token" -H "fiware-service: Tenant1" -H "fiware-servicepath: /" http://localhost:8000/v2/entities/urn:ngsi-ld:AirQualityObserved:demo/attrs/temperature`
+if [ $response == "200" ]
+then
+  echo "PASSED"
+else
+  echo "ERROR: Can create entity"
```
Comment (chicco785) [inline scripts/test_context_broker.sh:111]: Cannot read entity?

File: scripts/test_context_broker.sh [hunk 5]
```diff
@@ -134,113 +59,70 @@ curl -i --request POST 'http://localhost:8000/v2/entities' \
...
+echo ""
+
+echo "Can I read entity urn:ngsi-ld:AirQualityObserved:demo attributes?"
+echo "==============================================================="
+export response=`curl -s -o /dev/null -w "%{http_code}" -H "Authorization: Bearer $token" -H "fiware-service: Tenant1" -H "fiware-servicepath: /" http://localhost:8000/v2/entities/urn:ngsi-ld:AirQualityObserved:demo/attrs`
+if [ $response == "200" ]
+then
+  echo "PASSED"
+else
+  echo "ERROR: Can create entity"
```
Comment (chicco785) [inline scripts/test_context_broker.sh:98]: Cannot read entity attributes?

File: scripts/test_context_broker.sh [hunk 6]
```diff
@@ -2,18 +2,28 @@
...
+else
+  echo "ERROR: Can't read entities"
+  exit 1
+fi
 
+echo ""
 
-echo "\n\ncan i create an entity at / in Tenant1?"
-curl -i --request POST 'http://localhost:8000/v2/entities' \
+echo "Can I create an entity at / in Tenant1?"
```
Comment (chicco785) [inline scripts/test_context_broker.sh:24]: Can I create an entity in ServicePath / for Tenant1?

File: scripts/test_context_broker.sh [hunk 7]
```diff
@@ -2,18 +2,28 @@
...
-export token=`curl -d "client_id=client1&grant_type=password&username=admin&password=admin" -X POST 'http://keycloak:8080/auth/realms/master/protocol/openid-connect/token' | \
+export token=`curl -d "client_id=client1&grant_type=password&username=admin&password=admin" -X POST --header "Host: keycloak:8080" 'http://localhost:8080/auth/realms/master/protocol/openid-connect/token' | \
 jq '.access_token'`
 
 export token="${token%\"}"
 export token="${token#\"}"
 
-echo "\n\ncan i read entities?"
-curl -i -H "Authorization: Bearer $token" -H "fiware-service: Tenant1" -H "fiware-servicepath: /" 'http://localhost:8000/v2/entities'
+echo "Can i read entities?"
```
Comment (chicco785) [inline scripts/test_context_broker.sh:11]: Can i read all entities?

File: anubis-management-api/src/tests/test_policies.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Cerfoglg) [reply anubis-management-api/src/tests/test_policies.py:53]: Added now. Though I believe the WAC conversion should be expanded upon in another PR eventually

**Review by chicco785 — APPROVED** [review]

**Pull request merged**